pandas/_libs/tslibs/offsets.pyx

```diff
@@ -1986,52 +1986,88 @@ cdef class YearBegin(YearOffset):
 # Quarter-Based Offset Classes
 
 cdef class QuarterOffset(SingleConstructorOffset):
-    _attributes = tuple(["n", "normalize", "startingMonth"])
-    # TODO: Consider combining QuarterOffset and YearOffset __init__ at some
-    #       point.  Also apply_index, is_on_offset, rule_code if
-    #       startingMonth vs month attr names are resolved
+    """
+    Base for quarter-based offset classes.
+
+    Parameters
+    ----------
+    n : int, default 1
+        Number of quarters to offset.
+    normalize : bool, default False
+        If true, the time component of the resulting date-time is converted
+        to 00:00:00, i.e. midnight (the start, not the end of date-time).
+    month : int
+        The calendar month number (1 for January) of the beginning or ending
+        month in a custom-defined quarter to be used as an offset.
+
+        .. versionadded:: 1.3.0
+    startingMonth : int
+        The calendar month number (1 for January) of the beginning or ending
+        month in a custom-defined quarter to be used as an offset.
+
+        .. deprecated:: 1.3.0
+    """
 
-    # FIXME: python annotations here breaks things
-    # _default_starting_month: int
-    # _from_name_starting_month: int
+    _attributes = tuple(["n", "normalize", "month"])
+    # TODO: Consider combining QuarterOffset and YearOffset __init__ at some
+    #       point. Also apply_index, is_on_offset, rule_code.
 
     cdef readonly:
-        int startingMonth
+        int month
 
-    def __init__(self, n=1, normalize=False, startingMonth=None):
+    def __init__(self, n: int=1, normalize: bool=False, month: int=None,
```

Comment:
should we try to detect if month is passed as a non-keyword, in which case the behavior may be silently changing?

(if backward compat werent an issue, I'd be OK with making everything other than `n` and `normalize` keyword-only in all these classes)